npm-global-without-sudo.md

```diff
@@ -21,13 +21,28 @@ npm config set prefix "${HOME}/.npm-packages"
 Add the following to your `.bashrc`/`.zshrc`:
 
 ```sh
-NPM_PACKAGES="${HOME}/.npm-packages"
+NPM_PACKAGES="${XDG_DATA_HOME}/npm-packages"
 
-export PATH="$PATH:$NPM_PACKAGES/bin"
+if ! [ -d "$NPM_PACKAGES" ] && command -v npm >/dev/null 2>&1; then
+	npm config set prefix "$NPM_PACKAGES"
+	npm config set cache "${XDG_CACHE_HOME}/npm"
```

Comment:
Why are we overriding npm's defaults?

This also needs to support macOS.

Reply:
Because that is the point of this markdown file, to install it where no `sudo` is required for writing access. These commands above exist as a safety net  in case the user forgot to set them when setting up the system. For example, when transferring his configuration onto a new computer.

Reply:
This recipe is about "global binaries". I don't see why you need to change the npm cache location for that?

Reply:
Okay, the latest commit addreses this.

Reply:
Doesn't look like it's fixed.